src/plugins/recursion/README.md

```diff
@@ -0,0 +1,64 @@
+
+
+## Introduction
+
+I took [Relax NG](http://relaxng.org/spec-20011203.html#define-ref)'s `ref` and `define` part and tried to "copy" 
+their approach for elektra's specification.
```

Comment:
I like this idea! Iirc Relax NG is also mathematically sound, could you check if you find a paper with a proof?